File: AkitaModel/BgNhis/Examination.cs
namespace Skyware.Lis.AkitaModel.BgNhis;

// Ignore Spelling: lis

/// <summary>
/// NHIF covered examination (Изследване, платимо от НЗОК).
/// </summary>
public class Examination
{

    /// <summary>
    /// Code according to NHIF (НЗОК).
    /// </summary>
    public string NhifCode { get; set; }

    /// <summary>
    /// Flag, indicating ordering rules of NHIF (високо-специализирано изследване, ВСД)
    /// </summary>
    public bool IsSpecialized { get; set; }

    /// <summary>
    /// Code according to NHIS (НЗИС) set CL022.
    /// </summary>
    public string NhisCode { get; set; }

    /// <summary>
    /// Name of product according to NHIF (НЗОК).
    /// </summary>
    public string Name { get; set; }

    /// <summary>
    /// Mapped LOINC code, according to this laboratory
    /// </summary>
    public string LoincCode { get; set; }

    /// <summary>
    /// Id of the corresponding LIS test
    /// </summary>
    public int? TestId { get; set; }

    /// <summary>
    /// Id of the corresponding LIS profile/panel
    /// </summary>
    public int? ProfileId { get; set; }

    /// <summary>
    /// Examination name, according to laboratory
    /// </summary>
    public string LisName { get; set; }

}

File: AkitaMotelTS/src/BgNhis/Examination.ts
/**
 * Skyware.Lis.AkitaModel.BgNhis
 */

// Ignore Spelling: lis

/**
 * NHIF covered examination (Изследване, платимо от НЗОК).
 */
export class Examination {
  /**
   * Code according to NHIF (НЗОК).
   */
  nhifCode?: string;

  /**
   * Flag, indicating ordering rules of NHIF (високо-специализирано изследване, ВСД)
   */
  isSpecialized!: boolean;

  /**
   * Code according to NHIS (НЗИС) set CL022.
   */
  nhisCode?: string;

  /**
   * Name of product according to NHIF (НЗОК).
   */
  name?: string;

  /**
   * Mapped LOINC code, according to this laboratory
   */
  loincCode?: string;

  /**
   * Id of the corresponding LIS test
   */
  testId?: number;

  /**
   * Id of the corresponding LIS profile/panel
   */
  profileId?: number;

  /**
   * Examination name, according to laboratory
   */
  lisName?: string;
}
